packages/cryptography/src/PrivateKey.js

```diff
@@ -268,17 +270,25 @@ export default class PrivateKey extends Key {
             throw new Error("this private key does not support key derivation");
         }
 
-        const { keyData, chainCode } = await slip10.derive(
-            this.toBytesRaw(),
-            this._key._chainCode,
-            index
-        );
+        if (this._key instanceof Ed25519PrivateKey) {
+            const { keyData, chainCode } = await slip10.derive(
+                this.toBytesRaw(),
+                this._key._chainCode,
+                index
+            );
 
-        /** @type {new (bytes: Uint8Array, chainCode?: Uint8Array) => Ed25519PrivateKey | EcdsaPrivateKey} */
-        const constructor = /** @type {any} */ (this._key.constructor);
+            return new PrivateKey(new Ed25519PrivateKey(keyData, chainCode));
```

Comment:
Q: In `mneumonic` you calculate the keyPair with `nacl.sign.keyPair.fromSeed(keyData)`.
Here you use keyData directly.
Should they be consistent.

Reply:
This is because we use different libraries for `Ed25519` and `ECDSA` support. `NACL` aka `libsodium` doesn't support `ECDSA`. The library we use for `ECDSA` though, just provides us `keyData` and expects us to pass that around instead of some more concrete type.

Reply:
Okay